scripts/switch-yarn-to-pnpm.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set -euxo pipefail
+
+# Helper functions and setup of jq
+file_exists() {
+	[ -f "$1" ]
+}
+
+command_exists() {
+	command -v "$1" >/dev/null 2>&1
+}
+
+if ! command_exists jq; then
+	echo "jq is not installed. Attempting to install it..."
+	if command_exists apt-get; then
+		sudo apt-get update && sudo apt-get install -y jq
+	elif command_exists brew; then
+		brew install jq
+	else
+		echo "Error: Cannot install jq. Please install it manually and run this script again."
+		exit 1
+	fi
+fi
+
+cd repo
+
+if ! file_exists package.json; then
+	echo "ERROR: 'package.json' is missing. Cannot proceed with conversion."
+	exit 1
+fi
+
+if file_exists yarn.lock; then
+	echo "Removing yarn.lock..."
+	rm yarn.lock
+fi
+
+if file_exists package-lock.json; then
+	echo "Removing package-lock.json..."
+	rm package-lock.json
+fi
+
+echo "Replacing 'yarn' with 'pnpm' in package.json scripts..."
+jq 'if has("scripts") then .scripts |= with_entries(.value |= gsub("yarn "; "pnpm ")) else . end' package.json > temp.json && mv temp.json package.json
+
+echo "Updating engines clause in package.json..."
+jq '.engines = (.engines // {}) |
+    .engines *= {
+        "node": ">= 18",
+        "npm": "please-use-pnpm",
+        "yarn": "please-use-pnpm",
+        "pnpm": ">= 9"
+    }' package.json > temp.json && mv temp.json package.json
+
+echo "Installing dependencies with pnpm..."
+pnpm install
```

Comment:
Do we need a cleanup script for existing setups, like we have in core?

https://github.com/discourse/discourse/blob/main/.pnpmfile.cjs#L6C1-L19C1

That is: will `pnpm i` work in dev envs that have yarn-created node_modules dirs?

Reply:
The reason I added it to core was because pnpm's auto cleanup doesn't work in workspaces. Themes/plugins are much simpler, so I think it's probably fine to just let pnpm do its thing, and avoid duplicating the `.pnpmfile` hack in every single repo.

What do you think @cvx @tyb-talks ?